### PR TITLE
Support Mongo 3 on Xenial

### PR DESCRIPTION
**Pull request by mrgnr**
This is mostly a [PR from upstream](https://github.com/edx/configuration/pull/3731), plus a Gross Hack so that we can have Mongo auth enabled on standalone instances. The Gross Hack is required due to a [bug in Ansible](https://github.com/ansible/ansible/issues/22791) that I submitted a [patch](https://github.com/ansible/ansible/pull/22792) for. When the patch is merged and we upgrade to Ansible 2.3 or 2.4, we can get rid of the hack.

I tested this on standalone instances and replica sets, with auth enabled, with TLS.

_Beginning with Ficus, we will [use TLS for connections to Mongo](https://github.com/appsembler/configuration/pull/57)._ Certificates can be generated with [`ax create-tls-certs`](https://github.com/appsembler/ax/pull/12).

### CODE REVIEW

**Comment by mrgnr** [comment]
[(8) Ficus: Create `appsembler/ficus/develop` branch in configuration repo](https://trello.com/c/7ifnxKFQ/915-8-ficus-create-appsembler-ficus-develop-branch-in-configuration-repo)

**Comment by mrgnr** [comment]
[(4) Ficus: Configure Mongo3](https://trello.com/c/cMZujI2m/917-4-ficus-configure-mongo3)

**Comment by melvinsoft** [comment]
@mrgnr This is a bit beyond my knowledge, but I've checked the bug and ansible PR, and it looks good to me for now, but maybe the best is to wait for other review.

File: playbooks/library/mongodb_is_primary [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+EXAMPLES = '''
+- name: Get status for localhost node
+  mongodb_is_primary
+    host: localhost:27017
+    username: root
+    password: password
+
+'''
+# Magic import
+from ansible.module_utils.basic import *
```
Comment (johnbaldwin) [inline playbooks/library/mongodb_is_primary]: Not a fan of wildcard imports. Better to do `from ansible.module_utils import basic` Wonder if there is some issue that prevents `from ansible.module_utils import basic` from working. Not something we want to put effort into changing I supposed, but not happy about it

**Comment by johnbaldwin** [comment]
@mrgnr I'm in the same boat as @melvinsoft, but combed over it best with the knowledge I have. You're hack isn't too bad, it's straightforward.

So we only auth if not a mongo cluster? Do we do mongo clusters or always just standalone?

I didn't recognize anything to prevent us from moving forward with using this

File: playbooks/library/mongodb_is_primary (hunk 1 above)
Comment (mrgnr) [inline playbooks/library/mongodb_is_primary]: @johnbaldwin I'm not trying to cop out on this, but this is code I got from the upstream PR. In Ansible Land, this is the "correct" thing to do (I think that Ansible does some dirty hacks with import machinery that makes the `import *` necessary). Unfortunately, you'll see this "magic import" in any Ansible module that you look at.

**Comment by mrgnr** [comment]
@johnbaldwin We enable auth for both standalone and clustered deployments. I originally added auth in [a previous PR](https://github.com/appsembler/configuration/pull/38/files), but had to re-add it for the standalone case in this PR because the two config files have been merged into one, which caused the original change to be lost.

**Pull request merged**